Shadow DOM docs (#16509)
DataDog/documentation: content/en/real_user_monitoring/guide/shadow-dom.md

---
title: Enrich Your Session Replays With Shadow DOM Components
kind: guide
description: Guide about Shadow DOM compatibility with Session Replay.
further_reading:
- link: '/real_user_monitoring/session_replay/'
  tag: 'Documentation'
  text: 'Learn about Session Replay'
---

## Overview

Shadow DOM helps developers build more modern websites by allowing them to incorporate isolated and reusable components into their code. Often used in order to keep a clean code structure and avoid style clashes, Shadow DOM usage has become more prominent in modern web development practices. 

> Making sure you're able to effectively visualize all the elements which compose your pages in your replays is a key requirement that Datadog can fulfill. Datadog provides full support of Shadow DOM without any extra configuration required — components which are located inside Shadow DOM are automatically captured by Session Replay. In order to get started and have the benefit of enriched browser recordings, set up [Session Replay collection](https://docs.datadoghq.com/real_user_monitoring/session_replay/).

Reviewer: @acorretti just double checking here: it's not full support right? we don't support closed shadow DOM?

> Making sure you're able to effectively visualize all the elements which compose your pages in your replays is a key requirement that Datadog can fulfill. Datadog provides full support of Shadow DOM without any extra configuration required — components which are located inside Shadow DOM are automatically captured by Session Replay. In order to get started and have the benefit of enriched browser recordings, set up [Session Replay collection](https://docs.datadoghq.com/real_user_monitoring/session_replay/).

Reviewer 2: ```suggestion
Datadog provides support of open Shadow DOM without any extra configuration required — components which are located inside a shadow root are automatically captured by Session Replay.
```
With these changes it'd be less fluff, more insight.

> Making sure you're able to effectively visualize all the elements which compose your pages in your replays is a key requirement that Datadog can fulfill. Datadog provides full support of Shadow DOM without any extra configuration required — components which are located inside Shadow DOM are automatically captured by Session Replay. In order to get started and have the benefit of enriched browser recordings, set up [Session Replay collection](https://docs.datadoghq.com/real_user_monitoring/session_replay/).

Reviewer 3: > Datadog provides full support of Shadow DOM without any extra configuration

It's not true.

Who is the expected reader of this? A technical person?

As a developer I would like to know what is supported & what is not

> Making sure you're able to effectively visualize all the elements which compose your pages in your replays is a key requirement that Datadog can fulfill. Datadog provides full support of Shadow DOM without any extra configuration required — components which are located inside Shadow DOM are automatically captured by Session Replay. In order to get started and have the benefit of enriched browser recordings, set up [Session Replay collection](https://docs.datadoghq.com/real_user_monitoring/session_replay/).

Reviewer: the reader is a technical person so yes, i like the idea of listing what is not supported, good idea

---
title: Enrich Your Session Replays With Shadow DOM Components
kind: guide
description: Guide about Shadow DOM compatibility with Session Replay.
further_reading:
- link: '/real_user_monitoring/session_replay/'
  tag: 'Documentation'
  text: 'Learn about Session Replay'
---

## Overview

Shadow DOM helps developers build more modern websites by allowing them to incorporate isolated and reusable components into their code. Often used in order to keep a clean code structure and avoid style clashes, Shadow DOM usage has become more prominent in modern web development practices. 

> Starting with `v4.31.0` of the [RUM Browser SDK][1], Datadog provides support of open Shadow DOM without any extra configuration required — components which are located inside a shadow root are automatically captured by Session Replay.

Reviewer 4: ```suggestion
## Setup
Starting with `v4.31.0` of the [RUM Browser SDK][1], Datadog provides support of open Shadow DOM without requiring additional configuration. Components which are located inside a shadow root are automatically captured by Session Replay.
```

**Note**: The open Shadow DOM compatibility has been tested on popular frameworks.

> ## What is not supported

Reviewer 4: Similar to my feedback on the other PR, is it possible for us to instead list the things we do support?

Reviewer 5: @victorien-provenzano Are we planning on supporting the other modes? Maybe we could add an alert box at the top to highlight that only the open mode is supported. Or if we are not planning on supporting other modes, change the title to `Open Shadow DOM`

Author: @estherk15 thanks for the suggestion, went ahead and added an alert box!
